rasterio/_warp.pyx

```diff
@@ -479,11 +489,9 @@ def _reproject(
 
     # Clean up transformer, warp options, and dataset handles.
     finally:
-
-        if hTransformArg != NULL:
-            _gdal.GDALDestroyGenImgProjTransformer(hTransformArg)
-        if psWOptions != NULL:
-            _gdal.GDALDestroyWarpOptions(psWOptions)
+        _gdal.GDALDestroyGenImgProjTransformer(hTransformArg)
+        _gdal.GDALDestroyWarpOptions(psWOptions)
```

Comment:
@rouault is it okay to call these functions with NULL arguments? it was not clear to me that it is safe.